app/main.py

```diff
@@ -1,8 +1,25 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
+
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        Person.people[name] = self
 
 
 def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    person_list = []
+    for person_dict in people:
+        person = Person(person_dict["name"], person_dict["age"])
+        person_list.append(person)
+
+    for person in person_list:
+        for person_dict in people:
+            if person_dict["name"] == person.name:
+                if "wife" in person_dict:
+                    if person_dict["wife"] is not None:
```

Comment:
You can simplify these nested `if` statements by using logical operators. For example, `if person_dict.get('wife'):` would be more concise and avoid potential `KeyError` exceptions.